src/main/java/com/mg105/use_cases/ReplayGenerator.java

```diff
@@ -8,21 +8,35 @@ public class ReplayGenerator {
 
     //   No need for constructor as we would not initialize an instance for ReplayGenerator
 
-    public BattleCharacter AttributeInheritance(BattleCharacter battleCharacter)
-    {
+    /**
+     * let a new BattleCharacter that inherit all the previous player's attributes
+     *
+     * @param battleCharacter the battleCharacter in the previous game
+     * @return a new BattleCharacter that inherit all the previous player's attributes
+     */
+    public BattleCharacter AttributeInheritance(BattleCharacter battleCharacter) {
```

Comment:
Style: we use `camelCase` for functions in Java... not `PascalCase`. Note: you do this in muptiple places but I will only highlight the first one I see.